cogs/boolean_utils.py

```diff
@@ -0,0 +1,395 @@
+        formatted result by recursively considering every possible
+        combination of True and False for every variable and processing
+        the postfix expression with those inputs.
+        '''
+        self.result_formatted = ""
+        # Get all variables, sort them to ensure consistent tables between runs
+        vars = []
+        for elem in self.var_set:
+            vars.append(elem)
+        vars.sort()
+        var_num = 0
+        # This dictionary will keep track of the value of each variable
+        var_dict = {}
+        self.format_header(vars)
+
+        # Begin recursion
+        var_dict[vars[var_num]] = False
+        self.recurse_through_var(vars, var_dict, var_num + 1)
+        var_dict[vars[var_num]] = True
+        self.recurse_through_var(vars, var_dict, var_num + 1)
+
+        # Add bottom line of formatted result
+        self.result_formatted += "+" + "---+" * len(vars)
+        self.result_formatted += "-" * len(self.disp_exp) + "--+\n"
+
+    def recurse_through_var(self, vars, var_dict, depth):
+        '''
+        Recurse until every variable has been set to True or False, then
+        once the recursion depth is equal to the number of variables
+        process the expression by passing in the dictionary which is
+        keeping track of the value of each variable
+        '''
+        if depth >= len(self.var_set):
+            # Process expression for a single permutation of input
+            self.process_single_exp(var_dict, vars)
+        else:
+            # Continue recursion
+            var_dict[vars[depth]] = False
+            self.recurse_through_var(vars, var_dict, depth + 1)
+            var_dict[vars[depth]] = True
+            self.recurse_through_var(vars, var_dict, depth + 1)
+
+    def process_single_exp(self, dict_bool, vars):
+        '''
+        Processes a post-fix expression with the values of each variable
+        stroed in `dict_bool`.
+        '''
+        post_stack = []
+        for char in self.post_exp:
+            if char.isalpha():
+                # Append value of that variable to the stack
+                post_stack.append(dict_bool[char])
+            elif char in self.prec_dict:
+                # Compute result and add it back to stack
+                if char == '!':
+                    one = not post_stack.pop()
+                    post_stack.append(one)
+                if char == '*':
+                    one = post_stack.pop()
+                    two = post_stack.pop()
+                    result = one and two
+                    post_stack.append(result)
+                if char == '+':
+                    one = post_stack.pop()
+                    two = post_stack.pop()
+                    result = one or two
+                    post_stack.append(result)
+                if char == '^':
+                    one = post_stack.pop()
+                    two = post_stack.pop()
+                    result = one and not two or two and not one
```

Comment:
You *should* be able to do `bool(one) ^ bool(two)`

https://stackoverflow.com/questions/432842/how-do-you-get-the-logical-xor-of-two-variables-in-python

Reply:
Sure, why not =p